### ISSUE TITLE
Exporting the same variable with bats causes spurious SC2030 SC2031 warnings

### ISSUE THREAD

**Issue by jakub-bochenski** [koalaman/shellcheck#3263]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2030/SC2031
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

This looks similar to #2873

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bats

@test "test1" {
	export foo=bar

}


@test "test2" {
	export foo=baz

}
```

#### Here's what shellcheck currently says:

```
[Line 4:](javascript:setPosition(4, 16))
        export foo=bar
               ^-- [SC2030](https://www.shellcheck.net/wiki/SC2030) (info): Modification of foo is local (to subshell caused by @bats test).
 
[Line 10:](javascript:setPosition(10, 16))
        export foo=baz
               ^-- [SC2031](https://www.shellcheck.net/wiki/SC2031) (info): foo was modified in a subshell. That change might be lost.
```

#### Here's what I wanted or expected to see:

No warnings

**Comment by jakub-bochenski** [comment]
Ran into another way to trigger this. Instead of `export`ing the same variable name across two `@test` blocks (as in the report above), this one comes from **re-`source`-ing the same file** across `@test` blocks, and it ends up misattributing a totally unrelated `mapfile -t ARR < <(...)` assignment — populated by plain process substitution, not any real subshell — as happening in a subshell.

**ShellCheck version:** 0.11.0

### Reproduction

`lib.sh`:
```sh
fill_reply() {
  mapfile -t COMPREPLY < <(printf '%s\n' "$1")
}
```

`t.bats`:
```sh
#!/usr/bin/env bats

setup() {
  # shellcheck source=./lib.sh
  source "$BATS_TEST_DIRNAME/lib.sh"
}

@test "first test re-sources lib.sh mid-body" {
  # shellcheck source=./lib.sh
  source "$BATS_TEST_DIRNAME/lib.sh"
  true
}

@test "COMPREPLY is set by a plain mapfile, not a subshell" {
  fill_reply hello
  [ "${COMPREPLY[0]}" = hello ]
}
```

Run: `shellcheck -x t.bats`

```
In t.bats line 16:
  [ "${COMPREPLY[0]}" = hello ]
     ^-------------^ SC2031 (info): COMPREPLY was modified in a subshell. That change might be lost.
```

### Expected

No warning — `mapfile -t COMPREPLY < <(...)` runs `mapfile` in the current shell; only the command substitution's producer side is a subshell, and it never touches `COMPREPLY`.

### What makes it go away

Deleting the redundant `source` line from the first `@test` block (leaving `setup()`'s `source` as the only one) makes the warning disappear with no other change:

```sh
@test "first test does NOT re-source lib.sh mid-body" {
  true
}

@test "COMPREPLY is set by a plain mapfile, not a subshell" {
  fill_reply hello
  [ "${COMPREPLY[0]}" = hello ]
}
```

`shellcheck -x t2.bats` → clean, exit 0.
